filesystem: introduce swap

romain-dartigues opened the pull request:
* aggregate failures
* introduce swap support

##### SUMMARY

Before the patch (`/dev/DATA/swap` is an empty, newly created LV):

``` yaml
- name: mkswap
  filesystem: fstype=swap dev=/dev/DATA/swap
```

Returned:

```
ok: [docker02] => (item={u'fs_vfstype': u'swap', u'fs_spec': u'/dev/DATA/swap', u'size': u'2G'})
```

But on the target, nothing much happened:

``` console
root@docker02:~# blkid /dev/DATA/swap
root@docker02:~# file -Ls /dev/DATA/swap
/dev/DATA/swap: data
```

After the patch, the playbook says:

```
changed: [docker02] => (item={u'fs_vfstype': u'swap', u'fs_spec': u'/dev/DATA/swap', u'size': u'2G'})
```

Now on the target:

``` console
root@docker02:~# blkid /dev/DATA/swap
/dev/DATA/swap: UUID="40fc8cb4-f18a-4693-85e4-189b52ff66b3" TYPE="swap"
root@docker02:~# file -Ls /dev/DATA/swap
/dev/DATA/swap: Linux/i386 swap file (new style), version 1 (4K pages), size 524287 pages, no label, UUID=40fc8cb4-f18a-4693-85e4-189b52ff66b3
```

Thanks to the existing code, the swap functionality is idempotent and when re-run will detect the FS is already a swap:

```
ok: [docker02] => (item={u'fs_vfstype': u'swap', u'fs_spec': u'/dev/DATA/swap', u'size': u'2G'})
```

The target is:

```
No LSB modules are available.
Distributor ID: Ubuntu
Description:    Ubuntu 14.04.4 LTS
Release:        14.04
Codename:       trusty
```

##### ISSUE TYPE

 - Feature Pull Request
 - Bugfix Pull Request

##### COMPONENT NAME
- module filesystem

##### ANSIBLE VERSION
```
ansible 2.1.2.0
```

##### ADDITIONAL INFORMATION
Originally presented in ansible/ansible-modules-extras#3345; more information there.

robinro commented on lib/ansible/modules/system/filesystem.py:
```diff
@@ -110,13 +115,32 @@ def _get_fs_size(fssize_cmd, dev, module):
+                    # is in use, otherwise return default.
+                    block_count = block_size = 1
+
+        if rc:
```
This block should probably be unindented one level to work for all filesystem types.

romain-dartigues replied:
You are right, it's a typo on my side when I imported m initial patch... I'm fixing it, thank you for your eye.

calfonso commented:
@romain-dartigues Note: the bot has noted this PR requires a rebase, and we are waiting for someone from the community to come review this. You will want to rebase and fix any conflicts before having someone review the PR.

Akasurde commented:
@romain-dartigues Any news ?

romain-dartigues commented:
hope this will get somewhere...

romain-dartigues committed:
filesystem: swap (#23632)

romain-dartigues commented:
@Akasurde < finally got the grisp of the unittests; sorry for the verbose trial and errors, I couldn't test locally before pushing.

Do you think this PR could advance to the next stage?

Critics welcome.

romain-dartigues commented:
@robinro, @calfonso < hi there, do you think this PR could move on? I suppose the **need_revision** tag  is preventing it to get further...

Thanks.

gundalow commented:
@romain-dartigues Hi, you'll need to rebase this branch, see https://docs.ansible.com/ansible/latest/dev_guide/developing_rebasing.html

How to communicate with the Ansible community: https://docs.ansible.com/ansible/devel/community/communication.html
How to contribute to Ansible: https://docs.ansible.com/ansible/devel/community/index.html

romain-dartigues committed:
documentation errata

Swap should not override existing filesystems without force
because this case is taken care of in main() by `elif fs and not force:`.

romain-dartigues committed:
use minix as unsupported filesystem

gundalow commented:
Looks like this was done by @pilou- in https://github.com/ansible/ansible/pull/49456/

So I'm going to close this PR
